catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -524,12 +538,68 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+
+          final Metacard workspace = getWorkspaceMetacard(id);
+          Attribute queries =
+              workspace.getAttribute(WorkspaceConstants.WORKSPACE_ASSOCIATED_QUERIES);
+
+          if (queries != null) {
+            String[] queryIds =
+                queries.getValues().stream().map(String.class::cast).toArray(String[]::new);
+
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
```

Comment:
If the query metacard ID doesn't exist or it has been deleted verify that this code makes a best effort or doesn't fail